pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2024,6 +2029,22 @@ func (configgen *ConfigGeneratorImpl) appendListenerFallthroughRouteForCompleteL
 	}
 }
 
+// hash returns a unique hash based on listener.FilterChain, default to 32-bit CRC hash for performance considerations
+func hash(fc *listener.FilterChain) (res string, err error) {
+	var fcJSON []byte
+
+	h := crc32.New(crc32.MakeTable(crc32.Castagnoli))
+
+	if fcJSON, _ = json.Marshal(fc); err != nil {
```

Comment:
json marshaling the whole filter chain is not acceptable for performance. We used to do this, and it added a 10x overhead. Since we are only getting basic debugging improvement from this, I don't think its worth it.

Reply:
Thanks for the feedbacks!
Though comparing w/ other struct -> string solution, json marshaling has relatively better performance. Anyway, I would agree w/ the performance concerns and would like to close this one if no better solution.